File: packages/client/hmi-client/src/types/Document.ts
export type XDDArtifactProperties = {
	title: string;
	DOI: string;
	trustScore: string;
	abstractText: string;
	xddId: string;
	documentId: string;
	documentTitle: string;
	contentText: string;
	indexInDocument: number;
	contentJSON: { [key: string]: { [key: string]: string } };
	image: string;
	relevantSentences: string;
	sectionID: string;
	sectionTitle: string;
	caption: string;
	documentBibjson: DocumentType; // the embedded document metadata wherein this artifact is extracted
};

// XDD extraction object, which should match Extraction.java at the backend
export type XDDArtifact = {
	askemClass: string;
	properties: XDDArtifactProperties;
	askemId: string;
	xddCreated: Date;
	xddRegistrant: number;
	highlight: string[];
};

export type XDDUrlExtraction = {
	url: string;
	resourceTitle: string;
	extractedFrom: string[];
};

export type DocumentAuthor = {
	name: string;
};

export type DocumentIdentifier = {
	type: string;
	id: string;
};

export type DocumentLink = {
	type: string;
	url: string;
};

export type DocumentKnownTerms = {
	[term: string]: string[];
};

export type DocumentKnownEntity = {
	urlExtractions: XDDUrlExtraction[];
	summaries: {
		sections: { [key: string]: string };
	};
};

export type DocumentType = {
	abstractText: string; // mapped from abstract
	abstract: string;
	author: DocumentAuthor[];
	identifier: DocumentIdentifier[];
	journal: string;
	knownTerms?: DocumentKnownTerms[];
	link: DocumentLink[];
	number: string;
	pages: string;
	publisher: string;
	title: string; // name
	type: string;
	volume: string;
	year: string;
	gddId: string;
	highlight: string[];
	githubUrls: string[];
	knownEntities?: DocumentKnownEntity;

	// We don't know exactly what is in citationList - DC Jan 2023
	citationList: { [key: string]: string }[];

	// additional-client-side fields
	relatedDocuments?: Document[];
	relatedExtractions?: XDDArtifact[];
};

File: packages/services/hmi-server/src/main/java/software/uncharted/terarium/hmiserver/models/documentservice/Document.java
package software.uncharted.terarium.hmiserver.models.documentservice;

import com.fasterxml.jackson.annotation.JsonAlias;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.Accessors;

import java.io.Serializable;
import java.util.List;
import java.util.Map;

/**
 * XDD Document representation
 */
@Data
@Accessors(chain = true)
@NoArgsConstructor
public class Document implements Serializable {

	@JsonAlias("_gddid")
	private String gddId;

	private String title;

	@JsonAlias("abstract")
	private String abstractText;

	private String journal;

	private String type;

	private String number;

	private String pages;

	private String publisher;

	private String volume;

	private String year;

	private List<Map<String, String>> link;

	private List<Map<String, String>> author;

	private List<Map<String, String>> identifier;

	@JsonAlias("_github_urls")
	private List<String> githubUrls;

	@JsonAlias("known_terms")
	private Map<String, List<String>> knownTerms;

	@JsonAlias("_highlight")
	private List<String> highlight;

	@JsonAlias("related_documents")
	private List<Document> relatedDocuments;

	@JsonAlias("related_extractions")
	private List<Extraction> relatedExtractions;

	@JsonAlias("known_entities")
	private KnownEntities knownEntities;

	@JsonAlias("citation_list")
	private List<Map<String, String>> citationList;

	@JsonAlias("cited_by")
	private List<Map<String, Object>> citedBy;


}

